cdap-security/src/main/java/co/cask/cdap/security/store/FileSecureStore.java

```diff
@@ -0,0 +1,463 @@
+   */
+  @Override
+  public void put(String name, byte[] data, String description, Map<String, String> properties) throws IOException {
+    SecureStoreMetadata meta = SecureStoreMetadata.of(name, description, properties);
+    SecureStoreData secureStoreData = new SecureStoreData(meta, data);
+    Key key = null;
+    writeLock.lock();
+    try {
+      if (keyStore.containsAlias(name)) {
+        key = keyStore.getKey(name, password);
+      }
+      keyStore.setKeyEntry(name, new KeyStoreEntry(secureStoreData, meta), password, null);
+      // Attempt to persist the store.
+      flush();
+    } catch (KeyStoreException | NoSuchAlgorithmException | UnrecoverableKeyException e) {
+      // We failed to store the key in the key store. Throw an IOException.
+      throw new IOException("Failed to store the key. ", e);
+    } catch (IOException ioe) {
+      // The key was stored in the keystore successfully but we failed to flush it to the file system.
+      // Remove the key from the keystore and throw an IOException.
+      try {
+        if (key == null) {
+          deleteFromStore(name, password);
+        } else {
+          keyStore.setKeyEntry(name, key, password, null);
+        }
+      } catch (KeyStoreException | UnrecoverableKeyException | NoSuchAlgorithmException e) {
+        LOG.error("Failed to recover the store after a failed flush.");
+      }
+      throw ioe;
+    } finally {
+      writeLock.unlock();
+    }
+  }
+
+  private Key deleteFromStore(String name, char[] password) throws KeyStoreException,
+    UnrecoverableKeyException, NoSuchAlgorithmException {
+    writeLock.lock();
+    try {
+      Key key = null;
+      if (keyStore.containsAlias(name)) {
+        key = keyStore.getKey(name, password);
+        keyStore.deleteEntry(name);
+      }
+      return key;
+    } finally {
+      writeLock.unlock();
+    }
+  }
+
+  /**
+   * Deletes the element with the given name.
+   * @param name Name of the element to be deleted
+   */
+  @Override
+  public void delete(String name) throws IOException {
+    Key key = null;
+    writeLock.lock();
+    try {
+      key = deleteFromStore(name, password);
+      // Attempt to persist the store. If this fails then the keystore file will have an extra key
+      // that the in-memory store does not. This will be remedied the next time a flush happens.
+      // If another flush does not happen and the system is restarted, the only time that file is read,
+      // then we will have an extra key.
+      flush();
+    } catch (IOException ioe) {
+      if (key != null) {
+        try {
+          keyStore.setKeyEntry(name, key, password, null);
+        } catch (KeyStoreException e) {
+          throw new IOException("Failed to put back the key after a flush failure.", e);
+        }
+      }
+      throw ioe;
+    } catch (UnrecoverableKeyException | NoSuchAlgorithmException | KeyStoreException e) {
+      e.printStackTrace();
```

Comment:
Don't do `printStackTrace`. Isn't you suppose to wrap it with `IOException` like other methods do?